umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/iso_shared/collection_citation.clj

```diff
@@ -0,0 +1,178 @@
+
+(def creator-editor-xpath
+  (str "gmd:citedResponsibleParty/gmd:CI_ResponsibleParty"
+       "[gmd:role/gmd:CI_RoleCode/@codeListValue='author']"))
+
+(def publisher-release-place-xpath
+  (str "gmd:citedResponsibleParty/gmd:CI_ResponsibleParty"
+       "[gmd:role/gmd:CI_RoleCode/@codeListValue='publisher']"))
+
+(def online-resource-xpath
+  (str "gmd:citedResponsibleParty/gmd:CI_ResponsibleParty"
+       "[gmd:role/gmd:CI_RoleCode/@codeListValue='resourceProvider']"))
+
+(def title-xpath
+  "gmd:title/gco:CharacterString")
+
+(def series-name-xpath
+  "gmd:series/gmd:CI_Series/gmd:name/gco:CharacterString")
+
+(def release-date-xpath
+  "gmd:editionDate/gco:DateTime")
+
+(def delivery-point-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:deliveryPoint/gco:CharacterString")
+
+(def city-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:city/gco:CharacterString")
+
+(def state-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:administrativeArea/gco:CharacterString")
+
+(def zip-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:postalCode/gco:CharacterString")
+
+(def country-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:country/gco:CharacterString")
+
+(def email-xpath
+  "gmd:contactInfo/gmd:CI_Contact/gmd:address/gmd:CI_Address/gmd:electronicMailAddress/gco:CharacterString")
+
+(def version-xpath
+  "gmd:edition/gco:CharacterString")
+
+(def issue-identification-xpath
+  "gmd:series/gmd:CI_Series/gmd:issueIdentification/gco:CharacterString")
+
+(def data-presentation-form-xpath
+  "gmd:presentationForm/gmd:CI_PresentationFormCode/@codeListValue")
+
+(def other-citation-details-xpath
+  "gmd:otherCitationDetails/gco:CharacterString")
+
+(def linkage-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact"
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:linkage/gmd:URL")) 
+
+(def protocal-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact" 
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:protocol/gco:CharacterString")) 
+
+(def application-profile-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact" 
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:applicationProfile/gco:CharacterString"))
+
+(def name-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact" 
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:name/gco:CharacterString"))
+
+(def description-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact" 
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:description/gco:CharacterString"))
+
+(def function-xpath
+  (str "gmd:contactInfo/gmd:CI_Contact" 
+       "/gmd:onlineResource/gmd:CI_OnlineResource/gmd:function/gmd:CI_OnLineFunctionCode/@codeListValue"))
+
+(defn parse-collection-citation
+  "Parse the Collection Citation from XML resource citation"
+  [doc collection-citation-xpath sanitize?]
+  (let [resource-citations (seq (select doc collection-citation-xpath))]
+    (for [resource-citation resource-citations
```

Comment:
Thanks for pointing it out. I was planning to clean up the code after I got it working.  The reason I put it out there early is to catch the PI build error early.  Done.